### ISSUE TITLE
test-ended rule does not like macros:

### ISSUE THREAD

**Issue by jamestalmage** [avajs/eslint-plugin-ava#119]
``` js
function cbMacro(t, input, expected) {
  fn(input, output => {
     t.is(output, expected);
     t.end();
  });
}

test.cb(cbMacro, 'foo', 'FOO'); // <-- violates the rule, but it shouldn't.
test.cb(cbMacro, 'bar', 'BAR');
```

**Comment by IssuehuntBot** [comment]
@issuehunt has funded $20.00 to this issue.

---
- Submit pull request via [IssueHunt](https://issuehunt.io/repos/49722492/issues/119) to receive this reward.
- Want to contribute? Chip in to this issue via [IssueHunt](https://issuehunt.io/repos/49722492/issues/119).
- Checkout the [IssueHunt Issue Explorer](https://issuehunt.io/issues) to see more funded issues.
- Need help from developers? [Add your repository](https://issuehunt.io/r/new) on IssueHunt to raise funds.

**Comment by GMartigny** [comment]
How should the plugin react to this kind of case ?

I see two distinct possibilities:
```js
test.cb(macro);
```
```js
test.cb(t => {
  macro();
});
```

In the first case, we can "simply" disable the warning if `test.cb` first argument is a reference (we can at least check that the reference exists).

In the second case however, assuming that any function calls disable the rule renders it irrelevant. Most of the time (all the time ?), a `test.cb` will a call a function to test.

**Comment by novemberborn** [comment]
> In the first case, we can "simply" disable the warning if `test.cb` first argument is a reference (we can at least check that the reference exists).

This sounds good to me.

**Comment by IssuehuntBot** [comment]
@sindresorhus has rewarded $18.00 to @gmartigny. [See it on IssueHunt](https://issuehunt.io/repos/49722492/issues/119)

- :moneybag: Total deposit: $20.00
- :tada: Repository reward(0%): $0.00
- :wrench: Service fee(10%): $2.00
